src/main/java/org/hashids/Hashids.java

```diff
@@ -228,150 +243,154 @@ private String _encode(long... numbers) {
     for (int i = 0; i < numbers.length; i++) {
       numberHashInt += (numbers[i] % (i + 100));
     }
-    String alphabet = this.alphabet;
-    final char ret = alphabet.charAt((int) (numberHashInt % alphabet.length()));
+
+    char[] newAlphabet = alphabet;
+    final char ret = newAlphabet[(int) (numberHashInt % newAlphabet.length)];
 
     long num;
     long sepsIndex, guardIndex;
-    String buffer;
-    final StringBuilder ret_strB = new StringBuilder(this.minHashLength);
+    final StringBuilder buffer = new StringBuilder();
+    final StringBuilder ret_strB = new StringBuilder();
     ret_strB.append(ret);
     char guard;
 
     for (int i = 0; i < numbers.length; i++) {
       num = numbers[i];
-      buffer = ret + this.salt + alphabet;
+      buffer.setLength(0);
+      buffer.append(ret)
+          .append(salt)
+          .append(newAlphabet);
 
-      alphabet = Hashids.consistentShuffle(alphabet, buffer.substring(0, alphabet.length()));
-      final String last = Hashids.hash(num, alphabet);
+      newAlphabet = Hashids.consistentShuffle(newAlphabet, buffer.substring(0, newAlphabet.length).toCharArray());
+      final String last = Hashids.hash(num, newAlphabet);
 
       ret_strB.append(last);
 
       if (i + 1 < numbers.length) {
         if (last.length() > 0) {
-          num %= (last.charAt(0) + i);
-          sepsIndex = (int) (num % this.seps.length());
+          num %= last.charAt(0) + i;
+          sepsIndex = (int) (num % seps.length);
```

Comment:
always prefix methods/attributes with `this.`

Reply:
Why? This is not a JAVA rule. Perhaps a JS one.